src/Streamly/Internal/FileSystem/FD.hs

```diff
@@ -215,7 +214,7 @@ readArrayUpto :: Int -> Handle -> IO (Array Word8)
 readArrayUpto size (Handle fd) = do
     ptr <- mallocPlainForeignPtrBytes size
     -- ptr <- mallocPlainForeignPtrAlignedBytes size (alignment (undefined :: Word8))
-    withForeignPtr ptr $ \p -> do
+    unsafeWithForeignPtr ptr $ \p -> do
```

Comment:
We should perhaps keep it as `withForeignPtr`, `read` call below can throw exceptions.